src/sentry/static/sentry/app/views/settings/components/forms/projectMapperField.tsx

```diff
@@ -111,64 +164,146 @@ export class RenderField extends React.Component<RenderProps> {
               Option: customOptionProject,
               ValueContainer: customValueContainer,
             }}
-            ref={this.sentryProjectRef}
+            onChange={handleSelectProject}
+            value={selectedSentryProjectId}
           />
           <StyledSelectControl
-            placeholder={mappedDropdown.placeholder}
-            name="mappedDropwdown"
+            placeholder={mappedValuePlaceholder}
+            name="mappedDropdown"
             openMenuOnFocus
             options={mappedItemsToShow}
-            ref={this.mappedRef}
+            components={{
+              Option: customOptionMappedValue,
+              ValueContainer: customMappedValueContainer,
+            }}
+            onChange={handleSelectMappedValue}
+            value={selectedMappedValue}
           />
-          <Button
+          <StyledAddProjectButton
+            disabled={!selectedSentryProjectId || !selectedMappedValue}
             type="button"
             size="small"
-            label={t('Add')}
-            icon={<IconAdd />}
+            priority="primary"
             onClick={handleAdd}
-          />
-        </SelectContainer>
-      </Wrapper>
+          >
+            Add Project
+          </StyledAddProjectButton>
+          {nextUrl && (
+            <StyledBackToVercelButton
+              type="button"
+              size="small"
+              priority="primary"
+              icon="icon-exit"
+              href={nextUrl}
+              external
+            >
+              {nextButtonText}
+            </StyledBackToVercelButton>
+          )}
+        </Item>
+      </div>
     );
   }
 }
 
 const ProjectMapperField = (props: Props) => (
-  <InputField
+  <StyledInputField
     {...props}
+    resetOnError
+    inline={false}
+    stacked={false}
     field={(renderProps: RenderProps) => <RenderField {...renderProps} />}
   />
 );
 
 export default ProjectMapperField;
 
 const StyledSelectControl = styled(SelectControl)`
-  width: 50%;
+  width: 272px;
+  margin-left: ${space(1.5)};
 `;
 
-const SelectContainer = styled('div')`
+const Item = styled('div')`
+  margin: -1px;
+  border: 1px solid ${p => p.theme.gray400};
   display: flex;
+  align-items: center;
+  height: 60px;
 `;
 
-const Wrapper = styled('div')``;
+const ItemValue = styled('div')`
+  padding: ${space(0.5)};
+  margin-left: ${space(2)};
+`;
 
-const Item = styled('div')`
-  padding: 5px;
+const MappedItemValue = styled('div')`
+  display: flex;
+  padding: ${space(0.5)};
+  position: absolute;
+  left: 300px;
+`;
+
+const DeleteButton = styled(Button)`
+  position: absolute;
+  right: ${space(2)};
+`;
+
+const StyledIdBadge = styled(IdBadge)`
+  margin-left: ${space(3)};
+`;
+
+const StyledVercelIcon = styled(IconVercel)`
+  margin-right: ${space(0.5)};
+`;
+
+const StyledAddProjectButton = styled(Button)`
+  margin-left: ${space(2)};
+`;
+
+const StyledBackToVercelButton = styled(Button)`
+  margin-left: ${space(2)};
+`;
```

Comment:
Can this be laid out with a css grid instead of using absolute positioning + specific margins?

Reply:
@billyvg I spent a few minutes trying to use grid (never used it before) but couldn't get it looking remotely like what I wnated.